core/src/main/java/org/apache/gravitino/listener/TagEventDispatcher.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.listener;
+
+import java.util.Map;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.exceptions.NoSuchTagException;
+import org.apache.gravitino.tag.Tag;
+import org.apache.gravitino.tag.TagChange;
+import org.apache.gravitino.tag.TagDispatcher;
+
+public class TagEventDispatcher implements TagDispatcher {
+  @SuppressWarnings("unused")
+  private final EventBus eventBus;
+
+  @SuppressWarnings("unused")
+  private final TagDispatcher dispatcher;
+
+  public TagEventDispatcher(EventBus eventBus, TagDispatcher dispatcher) {
+    this.eventBus = eventBus;
+    this.dispatcher = dispatcher;
+  }
+
+  @Override
+  public String[] listTags(String metalake) {
+    // TODO: listTagsPreEvent
+    try {
+      // TODO: listTagsEvent
+    } catch (Exception e) {
+      // TODO: listTagFailureEvent
+      throw e;
+    }
+    return new String[0];
+  }
+
+  @Override
+  public Tag[] listTagsInfo(String metalake) {
+    // TODO: listTagsInfoPreEvent
+    try {
+      // TODO: listTagsInfoEvent
+    } catch (Exception e) {
+      // TODO: listTagsInfoFailureEvent
+      throw e;
+    }
+    return new Tag[0];
+  }
+
+  @Override
+  public Tag getTag(String metalake, String name) throws NoSuchTagException {
+    // TODO: getTagPreEvent
+    try {
+      // TODO: getTagEvent
+    } catch (NoSuchTagException e) {
+      // TODO: getTagFailureEvent
+      throw e;
+    }
+    return null;
```

Comment:
I think TagEventDispatcher should forward requests to the TagManager?

Reply:
+1, if not forwarding request, the tag request will takes no effect after this PR is merged.